### PR TITLE
Fix install script to handle debian 8.1 apt repo string

### LINKED ISSUE #14911: Installer script fails
Hey,

I want to inform you that your installer script (curl -sSL https://get.docker.com/ | sh) fails for debian 8.1.

The reason:

Your Code:

    if [ -z "$lsb_dist" ] && [ -r /etc/lsb-release ]; then
        lsb_dist="$(. /etc/lsb-release && echo "$DISTRIB_ID")"
        dist_version="$(. /etc/lsb-release && echo "$DISTRIB_CODENAME")"
    fi

The file which leads to the fail:

    # cat /etc/lsb-release
    DISTRIB_ID=Debian
    DISTRIB_RELEASE=8
    DISTRIB_CODENAME=
    DISTRIB_DESCRIPTION=

Whats wrong:

it copys "deb https://apt.dockerproject.org/repo debian- main" to repo ist, but it should be "deb https://apt.dockerproject.org/repo debian-jessie main"

### PR DESCRIPTION
**Pull request by runcom**
Fix #14911

Signed-off-by: Antonio Murdaca runcom@linux.com

### CODE REVIEW

**Comment by runcom** [comment]
ping @jfrazelle

**Comment by jfrazelle** [comment]
Sorry lsb-release isn't just Debian so I think we need something else ;)

**Comment by runcom** [comment]
right, I added a case for Debian/Ubuntu, don't know if this is enough tho

File: hack/install.sh [hunk 1]
```diff
@@ -124,6 +124,37 @@ do_install() {
 	if [ -z "$lsb_dist" ] && [ -r /etc/lsb-release ]; then
 		lsb_dist="$(. /etc/lsb-release && echo "$DISTRIB_ID")"
 		dist_version="$(. /etc/lsb-release && echo "$DISTRIB_CODENAME")"
+		if [ -z "$dist_version" ]; then
+			dist_version="$(. /etc/lsb-release && echo "$DISTRIB_RELEASE")"
+			case "$lsb_dist" in
+				"Debian")
+					case "$dist_version" in
```
Comment (jfrazelle) [inline hack/install.sh]: since we also have this on line 16 can we make it a sub function :innocent:

Comment (runcom) [inline hack/install.sh]: Sure thing

File: hack/install.sh [hunk 2]
```diff
@@ -124,6 +124,37 @@ do_install() {
...
+					;;
+				"Ubuntu")
+					case "$dist_version" in
+						"15.04")
+							dist_version="vivid"
+							;;
+						"14.04")
+							dist_version="trusty"
+							;;
+						"13.10")
```
Comment (jfrazelle) [inline hack/install.sh]: we dont support saucy ;)

Comment (runcom) [inline hack/install.sh]: So I'll update the doc as well cause I found it somewhere in the doc yesterday :/

Comment (jfrazelle) [inline hack/install.sh]: oh nice! thanks!

**Comment by runcom** [comment]
@jfrazelle updated

**Comment by jfrazelle** [comment]
ping @tianon

**Comment by tianon** [comment]
Seems fair -- this whole block of detection code is getting really, really hairy though :disappointed: (more impetus for deprecating this script entirely and just pointing to the official install docs).

LGTM

**Comment by calavera** [comment]
cherry picked into #15091

**Pull request merged**